LandSandBoat/server
# [AMK, Lua] Mission 13 puzzle 1 - Beaucedine Glacier runaround

**_I affirm:_**

## What does this pull request do?
This implements the first puzzle of mission 13 of AMK (see walkthrough [here](https://ffxiclopedia.fandom.com/wiki/A_Challenge!_You_Could_Be_a_Winner#)).  The goblin grenadier tells you to find pips, of which there are six, which based on the order of the vanadiel days of the week, draw a number between 0-9.  There is a odd bug that can occur where the args passed from the CS 507 are used to update the map, so the values sent through there can add erroneous pips to show up on the map.  This is solved by passing 0's through onEventUpdate, which leaves the map showing only what the player already discovered.  This function can also be used to change the map to show something else if desired... but unclear to what end at the moment.

## Steps to test these changes



`!addmission 10 12`
Zone into Castle Zvahl Bailey's and talk to shadowy pillar for cs
Go to lonely evergreen in beaucedine glacier for a cs
Talk to Goblin Grenadier in beaucedine glacier for puzzle instructions
Gather pips and determine answer according to wiki instructions
Return to goblin, ask for hints
Get wrong answer: chest should not open
Get correct answer: chest should open, get pocket mogbomb KI
- if you used 0 hints: 5 min flee, 1 hint: 3 min flee, 2 hints: no flee

Return to lonely evergreen for cs, which exchanges pocket mogbomb for trivia challenge kupon KI
